.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,55 @@
+name: build
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - id: checkout
+        name: Checkout
+        uses: actions/checkout@v3
+      - id: install-jhdk
+        name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - id: build
+        name: Build Docs
+        run: mvn -B clean verify
+      - id: create-gh-pages-artifact
+        name: Create Artifact
+        if: github.ref == 'refs/heads/main'
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: target/generated-docs
+  deploy:
+    name: Deploy to GitHub Pages
+    if: github.ref == 'refs/heads/main'
```

Comment:
unnecessary condition, this is always true for the currently configured 'on'

Reply:
Good catch!